packages/block-library/src/navigation/index.php

```diff
@@ -209,10 +209,22 @@ function block_core_navigation_build_html( $attributes, $block, $colors, $font_s
 			'<a class="wp-block-navigation-link__content"';
 
 		// Start appending HTML attributes to anchor tag.
-		if ( isset( $block['attrs']['url'] ) ) {
+		if (
+			isset( $block['attrs']['url'] ) &&
+			isset( $block['attrs']['type'] ) &&
+			'URL' === $block['attrs']['type']
+		) {
 			$html .= ' href="' . esc_url( $block['attrs']['url'] ) . '"';
 		}
 
+		if (
```

Comment:
It'd be good to avoid the possibility of two hrefs if another dev comes along later and makes a mistake with some of this logic. 

I thinkt the easiest option would be to use an `else` here.
```php
if (
  isset( $block['attrs']['url'] ) &&
  isset( $block['attrs']['type'] )
) {
    if ( 'URL' === $block['attrs']['type'] ) {
      // use the url attribute.
    } else {
      // get the permalink and use that.
    }
}
```